pkg/types/installconfig.go

```diff
@@ -154,6 +154,9 @@ type InstallConfig struct {
 	// +optional
 	Publish PublishingStrategy `json:"publish,omitempty"`
 
+	// InternalResources controls the visibility of ingress and apiserver. Defaults to public.
+	InternalResources *InternalResources `json:"internal,omitempty"`
```

Comment:
There's some discussion of this field in [this slack thread](https://redhat-internal.slack.com/archives/C011CSSPBLK/p1698240102565929), where @JoelSpeed suggested using a discriminated union for the `publish` field.  One way we could accomplish this is by adding a third `PublishingStrategy`, `Mixed`, and this field would only be configurable if `Mixed` publishing strategy is set.

Reply:
Yeah that's what I have to do, if I can only get this thing to work